fairworkflows/fairstep.py

```diff
@@ -425,16 +427,23 @@ def _modify_function(func):
         inputs = _extract_inputs_from_function(func, kwargs)
         outputs = _extract_outputs_from_function(func, kwargs)
 
-        func._fairstep = FairStep(uri='http://www.example.org/unpublished-'+func.__name__,
-                                  label=label,
-                                  description=description,
-                                  is_pplan_step=is_pplan_step,
-                                  is_manual_task=is_manual_task,
-                                  is_script_task=is_script_task,
-                                  language=LINGSYS_PYTHON,
-                                  inputs=inputs,
-                                  outputs=outputs)
-
+        fairstep = FairStep(uri='http://www.example.org/unpublished-'+func.__name__,
+                            label=label,
+                            description=description,
+                            is_pplan_step=is_pplan_step,
+                            is_manual_task=is_manual_task,
+                            is_script_task=is_script_task,
+                            language=LINGSYS_PYTHON,
+                            inputs=inputs,
+                            outputs=outputs)
+
+        def _add_logging(func):
+            def _wrapper(*func_args, **func_kwargs):
```

Comment:
Really minor thing: I wonder if we should use something like `@functools.wraps` here to make the function still 'look like' (i.e. same name etc) the original one. e.g. <https://docs.python.org/3/library/functools.html>. Seems to be common practice, anyway.

Reply:
Of course, thanks! I totally forgot that! (Done now)